packages/xo-web/src/xo-app/hub/index.js

```diff
@@ -0,0 +1,76 @@
+import _ from 'intl'
+import decorate from 'apply-decorators'
+import Icon from 'icon'
+import React from 'react'
+import { addSubscriptions } from 'utils'
+import { Container, Col, Row } from 'grid'
+import { injectState, provideState } from 'reaclette'
+import { isEmpty, map, mapValues, orderBy } from 'lodash'
+import { subscribeResourceCatalog } from 'xo'
+
+import Page from '../page'
+import Resource from './resource'
+
+// ==================================================================
+
+const DEFAULT_SORT_OPTION = {
+  labelId: 'hubSortByName',
+  sortBy: 'name',
+  sortOrder: 'asc',
+}
+
+const HEADER = (
+  <Container>
+    <h2>
+      <Icon icon='menu-hub' /> {_('hubPage')}
+    </h2>
+  </Container>
+)
+
+export default decorate([
+  addSubscriptions({
+    catalog: subscribeResourceCatalog({ filters: { hub: true } }),
+  }),
+  provideState({
+    initialState: () => ({
+      sortBy: DEFAULT_SORT_OPTION.sortBy,
+      sortOrder: DEFAULT_SORT_OPTION.sortOrder,
+    }),
+    computed: {
+      resources: ({ availableResources, sortBy, sortOrder }) =>
+        orderBy(availableResources, res => res[sortBy], sortOrder),
+      availableResources: (_, { catalog }) => {
+        if (catalog !== undefined) {
+          delete catalog._namespaces
+        }
+        return map(mapValues(catalog, 'xva'), (info, namespace) => ({
```

Comment:
```suggestion
        return map(catalog, (entry, namespace) => ({
          namespace,
          ...entry.xva,
        }))
```

Reply:
if `entry` doesn't have `xva` property, we will get `cannot destructure xva of undefined`

Reply:
You add a default value to `entry` => ` (entry = {}, namespace)`. This suggestion is to avoid `O(n²)`.

Reply:
Did you display entries without `xva` property?

Reply:
No, I display'em with `xva` property. Okey your suggestion is good to me :slightly_smiling_face:

Reply:
I think my suggestion is not good enough, we need also to filter entries without `xva` properties. I think construct an array instead of using `map` will be more appropriate.